src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFs.java

```diff
@@ -0,0 +1,334 @@
+
+        // take care of the VCF headers
+        final Set<VCFHeaderLine> headerLines = new LinkedHashSet<>(getHeaderForVariants().getMetaDataInInputOrder());
+        headerLines.addAll(annotationEngine.getVCFAnnotationDescriptions());
+        headerLines.addAll(genotypingEngine.getAppropriateVCFInfoHeaders());
+
+        // add headers for annotations added by this tool
+        headerLines.add(new VCFSimpleHeaderLine(GATKVCFConstants.SYMBOLIC_ALLELE_DEFINITION_HEADER_TAG, GATKVCFConstants.SPANNING_DELETION_SYMBOLIC_ALLELE_NAME_DEPRECATED, "Represents any possible spanning deletion allele at this location"));
+        headerLines.add(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.MLE_ALLELE_COUNT_KEY));
+        headerLines.add(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY));
+        headerLines.add(GATKVCFHeaderLines.getFormatLine(GATKVCFConstants.REFERENCE_GENOTYPE_QUALITY));
+        headerLines.add(VCFStandardHeaderLines.getInfoLine(VCFConstants.DEPTH_KEY));   // needed for gVCFs without DP tags
+        if ( dbsnp.dbsnp != null  )
+            VCFStandardHeaderLines.addStandardInfoLines(headerLines, true, VCFConstants.DBSNP_KEY);
+
+        final Set<String> sampleNameSet = samples.asSetOfSamples();
+        final VCFHeader vcfHeader = new VCFHeader(headerLines, sampleNameSet);
+        vcfWriter.writeHeader(vcfHeader);
+
+        logger.info("Notice that the -ploidy parameter is ignored in " + getClass().getSimpleName() + " tool as this is automatically determined by the input variant files");
+    }
+
+    @Override
+    public void apply(VariantContext vc, ReadsContext reads, ReferenceContext ref, FeatureContext features ) {
+        ref.setWindow(10,10);
+        final VariantContext mergedVC = ReferenceConfidenceVariantContextMerger.merge(Collections.singletonList(vc), vc, includeNonVariants ? ref.getBase() : null, true, false);
+        final VariantContext regenotypedVC = regenotypeVC(mergedVC, ref, features);
+        if (regenotypedVC != null) {
+            vcfWriter.add(regenotypedVC);
+        }
+    }
+
+    private static VariantContext removeNonRefSymbolicAllele(final VariantContext vc) {
+        final VariantContextBuilder builder = new VariantContextBuilder(vc);
+        final List<Allele> alleles = new ArrayList<>(vc.getAlleles());
+        alleles.remove(GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE);
+        builder.alleles(alleles);
+        return builder.make();
+    }
+
+
+    /**
+     * Re-genotype (and re-annotate) a combined genomic VC
+     * @return a new VariantContext or null if the site turned monomorphic and we don't want such sites
+     */
+    private VariantContext regenotypeVC(final VariantContext originalVC, final ReferenceContext ref, final FeatureContext features) {
+        Utils.nonNull(originalVC);
+
+        VariantContext result = originalVC;
+
+        // only re-genotype polymorphic sites
```

Comment:
extract separate methods for those snippets of code - here, comments are a bad smell and indicate how this big method should be chopped up